Lib/_pyio.py

```diff
@@ -2523,9 +2523,14 @@ def read(self, size=None):
                 size = size_index()
         decoder = self._decoder or self._get_decoder()
         if size < 0:
+
+            chunk = self.buffer.read()
+            if chunk is None:
+                raise BlockingIOError
```

Comment:
I found the error message under the C implementation is clear while for py it's empty and could be confusing as well. What about keeping the same error message to ensure the consistency between c & py?

Reply:
You raise a valid point. I'm not sure how I missed that in the Python implementation, but I've addressed it now and aligned the error message with the C version for consistency.